lib/segment/src/index/query_optimization/rescore_formula/formula_scorer.rs

```diff
@@ -155,24 +155,46 @@ impl FormulaScorer<'_> {
             }
             ParsedExpression::Sqrt(expr) => {
                 let value = self.eval_expression(expr, point_id)?;
-                Ok(value.sqrt())
+                let sqrt_value = value.sqrt();
+                if sqrt_value.is_nan() {
+                    // Undefined: Send this point to the bottom of results
+                    Ok(f32::NEG_INFINITY)
```

Comment:
Does this work for euclid as well, because we invert later?

Reply:
All score boosting will have `Order::LargerIsBetter`, this will be added to documentation. For the case of euclidean distance, users will have to negate the scores, to be explicit.